code/modules/organs/prosthetics/_prosthetics.dm

```diff
@@ -0,0 +1,196 @@
+/*
+  A system for easily and quickly removing your own bodyparts, with a view towards
+  swapping them out for new ones, or just doing it as a party trick to horrify an
+  audience. Current implementation only supports robolimbs and uses a modular_bodypart
+  value on the manufacturer datum, but I have tried to keep it generic for future work.
+  PS. jesus christ this was meant to be a half an hour port
+*/
+
+// External organ procs:
+// Does this bodypart count as a modular limb, and if so, what kind?
+/obj/item/organ/external/proc/get_modular_limb_category()
+	. = MODULAR_BODYPART_INVALID
+	if(BP_IS_PROSTHETIC(src) && model)
+		var/decl/prosthetics_manufacturer/manufacturer = decls_repository.get_decl(model)
+		if(!isnull(manufacturer?.modular_bodyparts))
+			. = manufacturer.modular_bodyparts
+
+// Checks if a limb could theoretically be removed.
+// Note that this does not currently bother checking if a child or internal organ is vital.
+/obj/item/organ/external/proc/can_remove_modular_limb(var/mob/living/carbon/human/user)
+	if(vital || !(limb_flags & ORGAN_FLAG_CAN_AMPUTATE))
+		return FALSE
+	var/bodypart_cat = get_modular_limb_category()
+	if(bodypart_cat == MODULAR_BODYPART_CYBERNETIC)
+		if(!parent_organ)
+			return FALSE
+		var/obj/item/organ/external/parent = user?.get_organ(parent_organ)
+		if(!parent || parent.get_modular_limb_category(user) < MODULAR_BODYPART_CYBERNETIC)
```

Comment:
Phrase as != unless there's actually some implied hierarchy here. Everything else is !=; may lead to random problems if someone adds a category.

Reply:
There is an implied hierarchy (as discussed on Disco) which is commented on the defines. The `<` is used because cybernetics and prosthetics share some behaviors with more stringent restrictions only applied to cybernetics.